File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    double n;
    int c = 0;
    //Prompt user for non negative float
    do
    {
        n = get_double("Change owed: ");
    }
    while (n < 0);

    //Converting dollars to cents
    n = n * 100;
    double round(double n);

    //Greedy algorithm
    //Decrease by quarters
    while (n >= 25)
    {
        c = c + 1 ;
        n = n - 25 ;
    }

    //Decrease by dimes
    while (n >= 10)
    {
        c = c + 1 ;
        n = n - 10 ;
    }

    //Decrease by nickels
    while (n >= 5)
    {
        c = c + 1 ;
        n = n - 5 ;
    }

    //Decrease by pennies
    while (n >= 1)
    {
        c = c + 1 ;
        n = n - 1 ;
    }

    printf("%i\n", c);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    n = get_float("Change owed: ")
    if n > 0:
        break

n = round(n * 100)
c = 0

while n >= 25:
    c += 1
    n -= 25

while n >= 10:
    c += 1
    n -= 10

while n >= 5:
    c += 1
    n -= 5

while n >= 1:
    c += 1
    n -= 1

print(c)